includes/classes/order.php

```diff
@@ -412,6 +415,7 @@ protected function cart(shoppingCart $basket, $customer_id = null, $sendto_addre
           $taxCountryId = $tax_address->fields['entry_country_id'];
           $taxZoneId = $tax_address->fields['entry_zone_id'];
         }
+        $this->notify('NOTIFY_ORDER_CART_TAX_ADDRESS', $tax_address);
```

Comment:
At this point in the processing, only the `$tax_address` is needed/recorded in the observer since that address defines the taxable location.  Special processing might be needed for certain "zones" (e.g. states, countries or zipcodes).